lv_objx/lv_chart.c

```diff
@@ -821,4 +886,248 @@ static void lv_chart_draw_vertical_lines(lv_obj_t * chart, const lv_area_t * mas
+
+        const lv_style_t * style = lv_obj_get_style(chart);
+        lv_opa_t opa_scale = lv_obj_get_opa_scale(chart);
+
+        uint8_t i,j;
+        uint8_t list_index;
+        uint8_t num_of_labels;
+        uint8_t num_scale_ticks;
+        uint8_t major_tick_len, minor_tick_len;
+        lv_point_t p1;
+        lv_point_t p2;
+        lv_coord_t x_ofs = chart->coords.x1;
+        lv_coord_t y_ofs = chart->coords.y1;
+        lv_coord_t h = lv_obj_get_height(chart);
+        lv_coord_t w = lv_obj_get_width(chart);
+        char buf[16+1]; /* up to 16 symbols per label + null terminator */
+
+        /* calculate the size of tick marks */
+        major_tick_len = (int32_t)w / 15;
+        minor_tick_len = major_tick_len * 2/3;
+
+        /* count the '\n'-s to determine the number of options */
+        list_index = 0;
+        num_of_labels = 0;
+        if(ext->y_axis.list_of_values != NULL)
+        {
+			for(j = 0; ext->y_axis.list_of_values[j] != '\0'; j++) {
+				if(ext->y_axis.list_of_values[j] == '\n')
+					num_of_labels++;
+			}
+
+	        num_of_labels++; /* last option in the at row*/
+        }
+
+        /* we can't have string labels without ticks step, set to 1 if not specified */
+        if(ext->y_axis.num_tick_marks == 0)
+        	ext->y_axis.num_tick_marks = 1;
+
+        /* calculate total number of ticks */
+        if(num_of_labels < 2)
+        	num_scale_ticks = ext->y_axis.num_tick_marks;
+        else
+        	num_scale_ticks = (ext->y_axis.num_tick_marks * (num_of_labels - 1));
+
+        for(i = 0; i < (num_scale_ticks + 1); i++ ) { /* one extra loop - it may not exist in the list, empty label */
+        	/* first point of the tick */
+            p1.x = 0 + x_ofs;
+
+        	/* second point of the tick */
+        	if( (num_of_labels != 0) && (i == 0 || i % ext->y_axis.num_tick_marks == 0) )
+        		p2.x = p1.x - major_tick_len; /* major tick */
+        	else
+        		p2.x = p1.x - minor_tick_len; /* minor tick */
+
+            /* draw a line at moving y position */
+        	p2.y = p1.y = y_ofs + h - (int32_t)(((int32_t)h * i) / num_scale_ticks + 1) - __LV_CHART_AXIS_Y_TICK_OFFSET_FIX;
+
+        	if( i != num_scale_ticks )
+        		lv_draw_line(&p1, &p2, mask, style, opa_scale);
+        	else
+        		if( (ext->y_axis.options & LV_CHART_AXIS_DRAW_LAST_TICK) != 0 )
+        			lv_draw_line(&p1, &p2, mask, style, opa_scale);
+
+            /* draw values if available */
+            if (num_of_labels != 0)
+            {
+            	/* add text only to major tick */
+            	if( i == 0 || i % ext->y_axis.num_tick_marks == 0 )
+            	{
+            		/* search for tick string */
+            		j = 0;
+            		while(	ext->y_axis.list_of_values[list_index] != '\n' &&
+            				ext->y_axis.list_of_values[list_index] != '\0')
+            		{
+            			/* do not overflow the buffer, but move to the end of the current label */
+            			if(j < 16)
```

Comment:
See my comment above about magic numbers. This also applies to the `x_ticks` variant.